ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -211,24 +226,149 @@ const DataProcConfigSelector = ({onChange, dataprocConfig})  => {
+const PresetSelector = ({hasMicroarrayData, setSelectedDiskSize, setSelectedMachine, setSelectedCompute, setSelectedDataprocConfig}) => {
+  {/* Recommended runtime: pick from default templates or change the image. */}
+  return <PopupTrigger side='bottom'
+                closeOnClick
+                content={
+                  <React.Fragment>
+                    {
+                      fp.flow(
+                        fp.filter(({runtimeTemplate}) => hasMicroarrayData || !runtimeTemplate.dataprocConfig),
+                        fp.toPairs,
+                        fp.map(([i, preset]) => {
+                          return <MenuItem
+                                style={styles.presetMenuItem}
+                                key={i}
+                                aria-label={preset.displayName}
+                                onClick={() => {
+                                  // renaming to avoid shadowing
+                                  const {runtimeTemplate} = preset;
+                                  const {presetDiskSize, presetMachineName, presetCompute} = fp.cond([
+                                    // Can't destructure due to shadowing.
+                                    [() => !!runtimeTemplate.gceConfig, (tmpl: Runtime) => ({
+                                      presetDiskSize: tmpl.gceConfig.diskSize,
+                                      presetMachineName: tmpl.gceConfig.machineType,
+                                      presetCompute: ComputeType.Standard
+                                    })],
+                                    [() => !!runtimeTemplate.dataprocConfig, ({dataprocConfig: {masterDiskSize, masterMachineType}}) => ({
+                                      presetDiskSize: masterDiskSize,
+                                      presetMachineName: masterMachineType,
+                                      presetCompute: ComputeType.Dataproc
+                                    })]
+                                  ])(runtimeTemplate);
+                                  const presetMachineType = fp.find(({name}) => name === presetMachineName, validLeonardoMachineTypes);
+
+                                  setSelectedDiskSize(presetDiskSize);
+                                  setSelectedMachine(presetMachineType);
+                                  setSelectedCompute(presetCompute);
+                                  setSelectedDataprocConfig(runtimeTemplate.dataprocConfig);
+                                }}>
+                              {preset.displayName}
+                            </MenuItem>;
+                        })
+                      )(runtimePresets)
+                    }
+                  </React.Fragment>
+                }>
+    <Clickable data-test-id='runtime-presets-menu'>
+      Recommended environments <ClrIcon shape='caret down'/>
+    </Clickable>
+  </PopupTrigger>;
+};
+
+const CostPredictor = ({
```

Comment:
opt: I like the word `Estimator` from your PR description slightly better than Predictor.